blocks/trending-news/trending-news.js

```diff
@@ -0,0 +1,172 @@
+  link.target = '_blank';
+  link.textContent = placeholders.discovermore;
+  const icon = document.createElement('i');
+  icon.className = 'icon-up-arrow icon';
+  link.appendChild(icon);
+  discoverMore.appendChild(link);
+  return discoverMore;
+}
+
+function createNewsCards(news) {
+  const article = document.createElement('div');
+  article.className = 'article';
+
+  const mediaWrapper = document.createElement('div');
+  mediaWrapper.className = 'picture-wrapper';
+
+  const picture = createPictureElement(news.imgUrl, 'article-thumbnail', false);
+
+  mediaWrapper.appendChild(picture);
+
+  const textContent = document.createElement('div');
+  textContent.className = 'article-text';
+
+  const h3 = document.createElement('h3');
+  h3.className = 'post-title';
+  const a = document.createElement('a');
+  a.href = news.link;
+  a.target = '_blank';
+  a.tabIndex = '0';
+  a.textContent = news.title;
+  h3.appendChild(a);
+  textContent.appendChild(h3);
+
+  const postMeta = document.createElement('div');
+  postMeta.className = 'post-info';
+  const iconSpan = document.createElement('span');
+  iconSpan.className = 'icon icon-icon-time';
+  const abbr = document.createElement('abbr');
+  abbr.textContent = `${news.date} `;
+  const abbrSource = document.createElement('abbr');
+  abbrSource.textContent = news.source;
+  postMeta.appendChild(iconSpan);
+  postMeta.appendChild(abbr);
+  postMeta.appendChild(abbrSource);
+  decorateIcons(postMeta);
+  textContent.appendChild(postMeta);
+
+  const description = document.createElement('div');
+  description.className = 'descriptn articleDesc';
+  textContent.appendChild(description);
+
+  article.appendChild(mediaWrapper);
+  article.appendChild(textContent);
+
+  return article;
+}
+
+export default function decorate(block) {
+  const newsData = getTrendingNews();
+  block.textContent = '';
+
+  const container = document.createElement('div');
+  container.className = 'container';
+
+  const titleWrap = document.createElement('div');
+  titleWrap.className = 'title text-center';
+  const h2 = document.createElement('h2');
+  h2.textContent = placeholders.trendingnews;
+  titleWrap.appendChild(h2);
+  container.appendChild(titleWrap);
+
+  const newsSection = document.createElement('div');
+  newsSection.className = 'news-section';
+
+  const slider = document.createElement('div');
+  slider.className = 'news-slider';
+
+  const newsTrack = document.createElement('div');
+  newsTrack.className = 'news-track';
+
+  newsData.forEach((news) => {
+    const slide = document.createElement('div');
+    slide.className = 'news-card';
+    const article = createNewsCards(news);
+    slide.appendChild(article);
+    newsTrack.appendChild(slide);
+  });
+
+  slider.appendChild(newsTrack);
+  newsSection.appendChild(slider);
+  newsSection.appendChild(createDiscoverMore());
+  container.appendChild(newsSection);
+
+  block.appendChild(container);
+  observe(block, createNewsCards);
```

Comment:
@ravverma  createNewsCards takes news as an argument. IMO those should be pass in observe as will in that case or should be calculate in createNewsCards method as well. @JiangLong2019  has updated the same in the pr https://github.com/aemsites/icicidirect/pull/64/files#diff-d02f5299c0b8a5f308bc088a03a095a0785cdd877862859f3bcf9a6a37266db6R91

Reply:
I am removing it for now and will do speratly after pr merge having all other changes as well